### ISSUE TITLE
Is it no longer possible to get previous/modified values in after events?

### ISSUE THREAD

**Issue by marbemac** [jagi/meteor-astronomy#338]
This was possible in v1, and is possible in most ORMs (mongoose, for example). It's useful so that one can take action after particular properties change, amongst other things.

**Comment by marbemac** [comment]
For reference, this is what we're doing right now, to expose previous properties:

```js
beforeSave({target: model}) {
    model._modified = model.getModifiedValues({old: true, raw: true});
}
```

IMHO this should be baked into astronomy.

**Comment by jagi** [comment]
I could add `getOriginal` method. I was also thinking about making such calls simpler. It could work the following way:
```js
doc.getOriginal(); // entire doc
doc.getOriginal('firstName'); // original "fieldName" before modifications
```
But it wouldn't take raw values

**Comment by marbemac** [comment]
That would be perfect!

**Comment by lukejagodzinski** [comment]
I don't know if it's gonna appear in 2.0. Because if I will be adding more and more features that people still want it will be postponed even more. So first I want to finish documentation, release 2.0 and probably in 2.0.1 introduce such method. We will see.

**Comment by Andrew1431** [comment]
If you had to ballpark your 2.0 release, what would be your best guess?

**Comment by lukejagodzinski** [comment]
@Andrew1431 you mean release date? If so, then it should be the end of this month

**Comment by lukejagodzinski** [comment]
@marbemac I don't know how I could missed it that you were talking about the `after` events. And as you know in the after events it's not so easy to access previous values. I understand that in some situations it may be handy but it will be hard to implement. As you wrote, right now the best way to do it is just to store old modified values in the `_modified` property of a document. But that's not a perfect solution because it's static. What if a developer what's non-raw values? So instead we should store entire document before modifications. But, we will end up having the `_original` property in a document which is coming back to how it was done in Astronomy 1.0. And I don't want such properties on the level of a document. However, there is one solution that I could try. Make the `_original` property only temporary, just for the save operation. I will test it out.

**Comment by lukejagodzinski** [comment]
Sorry but I can't do it. There are to many little problems that such a feature would create. Moreover it would significantly slow down the save operation and I don't want it. I know that your solution is not perfect and beautiful but it works and I think it's the best for now.

The other solution I think of that I can think about implementing after 2.0 release is ability to pass some data between events. So in the `beforeSave` event you can store in an event object some that and it would be visible in the `afterSave` event. But that would require more time, so I will postpone it.

**Comment by marbemac** [comment]
Sounds good! We can live with the solution above for now.

**Comment by s7dhansh** [comment]
@lukejagodzinski was this changed sometime? I seem to have access to isModified, getModifiedValues etc  in afterSave.
Is it a standard behavior, and should we use them in after events?

**Comment by lukejagodzinski** [comment]
@s7dhansh you can always access `isModified` and `getModifiedValues` methods from whatever event you want. The only difference is that in the `afterSave` event, the document is already saved so it will return a different value than in the `beforeSave` event. There was no change recently.

**Comment by s7dhansh** [comment]
@lukejagodzinski oh! so only getting old values by `getModifiedValues({old: true})` will not work, but `getModifiedValues()` will give the list of modified values. Am I correct?

Also, `isModified` is true for only top level fields. So it will be true for `a` but not `a.b`, in after events. Can  you kindly explain this, I am a bit confused here.

**Comment by lukejagodzinski** [comment]
@s7dhansh no, none of them, in the `afterSave` event, will work as they work in the `beforeSave` event. The state of the document changes after save so all the function will return different values.
